sample-apps/discounts-tutorial/web/frontend/hooks/useUpdateDiscount.js

```diff
@@ -0,0 +1,64 @@
+import { DiscountMethod } from '@shopify/discount-app-components';
+import { gql } from 'graphql-request';
+import { idToGid } from '../utilities/gid';
+
+import { useShopifyMutation } from './useShopifyMutation';
+
+const UPDATE_AUTOMATIC_MUTATION = gql`
+  mutation UpdateDiscount(
+    $id: ID!
+    $discount: DiscountAutomaticAppInput!
+  ) {
+    discountUpdate: discountAutomaticAppUpdate(id: $id, automaticAppDiscount: $discount) {
+      userErrors {
+        code
+        message
+        field
+      }
+    }
+  }
+`;
+
+const UPDATE_CODE_MUTATION = gql`
+  mutation UpdateDiscount(
+    $id: ID!
+    $discount: DiscountCodeAppInput!
+  ) {
+    discountUpdate: discountCodeAppUpdate(id: $id, codeAppDiscount: $discount) {
+      userErrors {
+        code
+        message
+        field
+      }
+    }
+  }
+`;
+
+export function useUpdateDiscount(discountMethod) {
```

Comment:
In a similar vein to my comments in `useDeleteDiscount`, I think the discount information should either all be passed in here or the callback, rather than split across the two.